src/EditorFeatures/Core/Editor/TextEditApplication.cs

```diff
@@ -17,7 +17,7 @@ internal static void UpdateText(SourceText newText, ITextBuffer buffer, EditOpti
         var oldSnapshot = buffer.CurrentSnapshot;
         var oldText = oldSnapshot.AsText();
         var changes = newText.GetTextChanges(oldText);
```

Comment:
>GetTextChanges [](http://example.com/codeflow?start=30&length=14)

I took a peek at SourceText.GetTextChanges and this is a bit orthogonal, but it feels like it could use a bit of love from your fancy new FixedSizeArrayBuilder. 

Also, I feel like it should have returned an ImmutableArray instead of an IReadOnlyList, but it's public, so this is a moot thought.

Reply:
Yes and yes.  I'm not touching compiler side for now.  And i can't touch public APIs.  My PRs are all about niceness that can happen in bulk :)